gazebo_grasp_plugin/src/GazeboGraspFix.cpp

```diff
@@ -559,43 +562,71 @@ void GazeboGraspFix::OnUpdate() {
         {
             CollidingPoint& cpInfo=pointIt->second;
             // initial distance from link to contact point (relative to link)
-            gazebo::math::Vector3 relContactPos=cpInfo.pos/cpInfo.sum;
+            gz_math_Vector3d relContactPos=cpInfo.pos/cpInfo.sum;
             // initial distance from link to object (relative to link)
-            gazebo::math::Vector3 relObjPos=cpInfo.objPos/cpInfo.sum;
+            gz_math_Vector3d relObjPos=cpInfo.objPos/cpInfo.sum;
            
-            // get current world pose of object 
-            gazebo::math::Pose currObjWorldPose=cpInfo.collObj->GetLink()->GetWorldPose();
+            // get current world pose of object
+#if GAZEBO_MAJOR_VERSION >= 8
+            gz_math_Pose3d currObjWorldPose=cpInfo.collObj->GetLink()->WorldPose();
```

Comment:
To reduce code duplication it would be better to have helper functions like getters and setters, in this style:

```suggestion
            GzPose3d currObjWorldPose = GetPose(cpInfo.collObj->GetLink());
            
            ///... and the getter function
            GzPose3d GetPose(const LinkPtr link)
            {
            #if GAZEBO_MAJOR_VERSION >= 8
              return link->WorldPose();
            #else 
             return link->GetWorldPose();
            #endif 
             }
```

then the code here can be kept free of  `GAZEBO_MAJOR_VERSION`.